src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java

```diff
@@ -110,4 +135,77 @@ protected static List<Configuration> getModuleConfigsByIds(String... moduleIds)
         return getModuleConfigsByIds(CONFIGURATION, moduleIds);
     }
 
+    /**
+     * Performs verification of the file with the given file path against the whole google config.
+     *
+     * @param filePath file path to verify.
+     * @throws Exception if exception occurs during verification process.
+     */
+    public void verifyWithWholeGoogleConfig(String filePath) throws Exception {
+        final Configuration[] children = CONFIGURATION.getChildren();
+        final DefaultConfiguration root = new DefaultConfiguration(ROOT_MODULE_NAME);
+
+        for (Configuration child : children) {
+            root.addChild(child);
+        }
+
+        verifyWithItConfig(root, filePath);
+    }
+
+    /**
+     * Performs verification of the file with the given file path against google config.
+     * It uses the specified list of modules to load them from google config for validation.
+     *
+     * @param listOfModules list of modules to load from google config.
+     * @param filePath file path to verify.
+     * @throws Exception if exception occurs during verification process.
+     */
+    public final void verifyWithGoogleConfigParser(String[] listOfModules,
+           String filePath) throws Exception {
+        final DefaultConfiguration googleConfig = new DefaultConfiguration(ROOT_MODULE_NAME);
+        final DefaultConfiguration treeWalkerConfig =
+                new DefaultConfiguration(TreeWalker.class.getSimpleName());
+
+        for (String module : listOfModules) {
+            final List<Configuration> children =
+                    getConfigChildren(module);
+            if (CHECKER_CHILDREN.contains(module)) {
+                for (Configuration child : children) {
+                    googleConfig.addChild(child);
+                }
+            }
+            else {
+                for (Configuration child : children) {
+                    treeWalkerConfig.addChild(child);
+                }
+            }
+        }
+
+        if (treeWalkerConfig.getChildren().length > 0) {
+            googleConfig.addChild(treeWalkerConfig);
+        }
+
+        verifyWithItConfig(googleConfig, filePath);
+    }
+
+    /**
+     * Gets the specified children module(s) from the config.
+     *
+     * @param module the children module to get. It can be either a module name or module id.
+     * @return list of children modules.
+     * @throws CheckstyleException if there is a problem for getting the module.
+     */
+    private static List<Configuration> getConfigChildren(String module)
```

Comment:
This looks like it should be in `AbstractItModuleTestSupport`. I am not seeing anything that says "Google".

Reply:
It is used inside verifyWithGoogleConfigParser() :

https://github.com/checkstyle/checkstyle/blob/aff00e1ab05cc59fc35158b99cf380ae1cabe3ca/src/it/java/com/google/checkstyle/test/base/AbstractGoogleModuleTestSupport.java#L169-L172

It can't be moved to IT abstract class

Reply:
Can't `verifyWithGoogleConfigParser` still access it when we switch the method to `protected`? I am still not seeing an issue.

Reply:
Yes it will be accessible. I kept it public because all other `verifyWithxxxxx` methods in `AbstractModuleTestSupport` are also public. 

I'll change this.

Reply:
I am going to drop this move for now.

I still think `AbstractItConfig` is the way to go, but it is more work than we need right now. It will have to be revisted when Sun is updated.

Reply:
@Zopsss This doesn't look completely done.

> final DefaultConfiguration googleConfig = new DefaultConfiguration(ROOT_MODULE_NAME);

Reply:
This one is resolved now.